test/middleware/company.authentication.middleware.unit.ts

```diff
@@ -27,40 +36,36 @@ describe("company authentication middleware tests", () => {
 
   beforeEach(() => {
     mockCompanyAuthMiddleware.mockClear();
-    next.mockClear();
-    req.originalUrl = URL;
+    mockSessionMiddleware.mockClear();
+    mockServiceAvailabilityMiddleware.mockClear();
+    mockAuthenticationMiddleware.mockClear();
+    mockLoggerErrorRequest.mockClear();
   });
 
-  it("should call CH authentication library when company pattern in url", () => {
-    companyAuthenticationMiddleware(req, res, next);
+  it("should call CH authentication library when company pattern in url", async () => {
+    await request(app).get(URL);
```

Comment:
i thought it would be better to test the param extraction by sending a 'real' url rather then call middleware directly